### PR TITLE
Revise Imu SDD and unit tests

### LINKED ISSUE #21: Improve IMU unit tests
The IMU unit tests in general look good. Fixing a couple of issues would make them better.

1. This code

    https://github.com/fprime-community/fprime-system-reference/blob/8ba7045b9d966fcc79656a79fef0b11d39d2126c/SystemReference/Gnc/Imu/test/ut/Tester.cpp#L42-L44

    asserts that some random test input values are not equal to some other values. We shouldn't make assertions that compare random values. Similarly for the code in `testGetGyroTlm`.

2. We could fix it by asserting that the values returned are the same values stored in the component. A more robust fix would be to keep track of the values sent in by the test handler functions, and verify that the same values are coming out.

### PR DESCRIPTION
**Pull request by bocchino**
* Revise and expand the SDD
* Improve the unit tests

Closes #21.

### CODE REVIEW

File: SystemReference/Gnc/Imu/Imu.hpp [hunk 1]
```diff
@@ -19,7 +19,16 @@ class Imu : public ImuComponentBase {
 
     // Values for the static consts were attained from the data sheet of the
     // MPU6050
-    static const U8 I2C_DEV0_ADDR = 0x68;
+
+    //! The I2C device addresses
+    struct I2cDevAddr {
```
Comment (LeStarch) [inline SystemReference/Gnc/Imu/Imu.hpp:24]: We really ought to call this `ImuDevAddr`.  The I2C protocol is clear addresses at `U8`, however; it is the IMU that specifies the choice of 2 addresses.  I do like the revised structure!

File: SystemReference/Gnc/Imu/docs/sdd.md [hunk 2]
```diff
@@ -2,34 +2,61 @@
...
+`Gnc::Imu` is an F' passive component that collects data from the MPU6050 6-DoF Accelerometer and Gyro. 
 
-## 2. Assumptions
-The design of `Imu` assumes the following:
-1. Data collected by the Imu arrives through a pull interface
+### 1.1. Hardware Overview
+
+**I2C interface:**
+The sensor reports data via an I2C interface.
+I2C data bytes are eight bits wide. The high-order bit is a
```
Comment (LeStarch) [inline SystemReference/Gnc/Imu/docs/sdd.md:11]: I believe this only applies to the address byte.  I think the data bytes are allowed the full range of 0-255.

File: SystemReference/Gnc/Imu/docs/sdd.md [hunk 3]
```diff
@@ -41,59 +68,120 @@ The diagram below shows the `Imu` component.
...
+1. `m_gyro`: A value of type `Gnc::ImuData` that stores the latest gyroscope data
+2. `m_accel`: A value of type `Gnc::ImuData` that stores the latest acceleration data
+3. `m_i2cDevAddress`: A value of type `U8` that stores the address of the MPU6050 sensor
+4. `m_setup`: A value of type `bool` that indicates whether the sensor has been activated
+
+### 4.6. Runtime Configuration
+At startup, the F Prime software must call the `setup` method of the
+`Imu` object.
+```
+void setup(I2cDevAddr devAddress);
```
Comment (LeStarch) [inline SystemReference/Gnc/Imu/docs/sdd.md:117]: `I2cDevAddr::T`

**Review by LeStarch — CHANGES_REQUESTED** [review]
@bocchino this is a very nice refactor.  Just had a few comments.

File: SystemReference/Gnc/Imu/Imu.hpp (hunk 1 above)
Comment (bocchino) [reply SystemReference/Gnc/Imu/Imu.hpp:24]: The full name of the type is `Gnc::Imu::I2cDevAddr`. Does this address the concern?

File: SystemReference/Gnc/Imu/docs/sdd.md (hunk 2 above)
Comment (bocchino) [reply SystemReference/Gnc/Imu/docs/sdd.md]: Fixed.

File: SystemReference/Gnc/Imu/docs/sdd.md (hunk 3 above)
Comment (bocchino) [reply SystemReference/Gnc/Imu/docs/sdd.md]: Fixed.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**